src/pkg/services/m365/api/lists.go

```diff
@@ -290,25 +498,67 @@ func CloneListItem(orig models.ListItemable) models.ListItemable {
 func retrieveFieldData(orig models.FieldValueSetable) models.FieldValueSetable {
 	fields := models.NewFieldValueSet()
 	additionalData := make(map[string]any)
-	fieldData := orig.GetAdditionalData()
-
-	// M365 Book keeping values removed during new Item Creation
-	// Removed Values:
-	// -- Prefixes -> @odata.context : absolute path to previous list
-	// .           -> @odata.etag : Embedded link to Prior M365 ID
-	// -- String Match: Read-Only Fields
-	// -> id : previous un
-	for key, value := range fieldData {
-		if strings.HasPrefix(key, "_") || strings.HasPrefix(key, "@") ||
-			key == "Edit" || key == "Created" || key == "Modified" ||
-			strings.Contains(key, "LookupId") || strings.Contains(key, "ChildCount") || strings.Contains(key, "LinkTitle") {
-			continue
-		}
 
-		additionalData[key] = value
+	if orig != nil {
+		fieldData := orig.GetAdditionalData()
+
+		// M365 Book keeping values removed during new Item Creation
+		// Removed Values:
+		// -- Prefixes -> @odata.context : absolute path to previous list
+		// .           -> @odata.etag : Embedded link to Prior M365 ID
+		// -- String Match: Read-Only Fields
+		// -> id : previous un
+		for key, value := range fieldData {
+			_, isReadOnlyField := readOnlyFieldNames[key]
+
+			if strings.HasPrefix(key, "_") ||
+				strings.HasPrefix(key, "@") ||
+				isReadOnlyField ||
+				strings.Contains(key, "LinkTitle") ||
+				strings.Contains(key, "ChildCount") {
+				continue
+			}
+
+			additionalData[key] = value
+		}
 	}
 
+	retainPrimaryAddressField(additionalData)
+
 	fields.SetAdditionalData(additionalData)
 
 	return fields
 }
+
+func retainPrimaryAddressField(additionalData map[string]any) {
+	if !hasAddressFields(additionalData) {
+		return
+	}
+
+	for k := range readonlyAddressFieldNames {
+		delete(additionalData, k)
+	}
+}
+
+func hasAddressFields(additionalData map[string]any) bool {
+	for key, value := range additionalData {
+		if nestedFields, ok := value.(map[string]any); ok &&
+			hasRequiredFields(nestedFields, addressFieldNames) &&
+			hasRequiredFields(additionalData, readonlyAddressFieldNames) &&
+			key != "GeoLoc" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func hasRequiredFields(data map[string]any, checkFieldNames map[string]struct{}) bool {
```

Comment:
Suggestion: this seems like a helpful func to have in a common package.  something like `/common/map` with the signature:
```go
func HasKeys(data map[string]any, keys ...string) bool {
```

That would both give it a central location, and make it something that's easy to test independent of the api code.